### ISSUE TITLE
Python bindings for stability checks

### ISSUE THREAD

**Issue by xdvriend** [GQCG/GQCP#829]
**Describe the feature you'd like**
The stability analysis is now implemented in GQCP. The next step would be to be able to perform these calculations using the python bindings as well.

**Additional context**
This is part 1 of the stability analysis python bindings issue #794 . Once issue #793 is completed, #794 can be finished completely.

This part of issue #794 wasn't really blocked by the ability to follow internal instabilities. Hence why I split up #794 into two separate issues and changed the original issue to a small epic.

**Comment by xdvriend** [comment]
@lelemmen I would like to tackle this implementation, of course following the newly adapted style of using an interface in the python bindings in order to avoid code duplication, since my sprint is currently on hold. This would entail refactoring the `QCModel::R/U/GHF` bindings as well, which would be a part of #811.
